ansible_events_ui/rbac.py

```diff
@@ -0,0 +1,30 @@
+from sqlalchemy.ext.asyncio import AsyncSession
+from .db.models import User, actions, role_actions, user_roles, roles
+from sqlalchemy import select
+from fastapi import HTTPException
+
+
+async def query_action(db: AsyncSession, user: User, action: str):
+
+    result = await db.execute(
+        select(actions.c.id, actions.c.name).select_from(
+            actions.join(role_actions).join(roles).join(user_roles).join(User)
+        ).where(User.email == user.email, actions.c.name == action)
+    )
```

Comment:
Minor: I think it's better for the readability to keep the structure flat:

```suggestion
    result = await db.execute(
        select(actions.c.id, actions.c.name)
        .select_from(actions)
        .join(role_actions)
        .join(roles)
        .join(user_roles)
        .join(User)
        .where(User.email == user.email, actions.c.name == action)
    )
```